docs/examples/tutorial/pure/cast.py

```diff
@@ -0,0 +1,11 @@
+import cython
+
+T = cython.typedef(cython.p_int)
+
+cython.declare(t=cython.uint)
+
+t = 12345
+
+t1 = cython.cast(cython.size_t, cython.cast(T, t))
+
+t2 = cython.cast(cython.size_t, cython.cast(T, t, typecheck=True))
```

Comment:
I'm not sure if this is the best example - `typecheck` would only really do something useful on a Python type - it has no way of checking if something really is an `int*` for example